cgroups/src/v1/memory.rs

```diff
@@ -101,11 +101,12 @@ impl Controller for Memory {
         Ok(())
     }
 
-    fn needs_to_handle<'a>(controller_opt: &'a ControllerOpt) -> Option<&'a Self::Resource> {
-        if let Some(memory) = &controller_opt.resources.memory {
-            return Some(memory);
-        }
-
+    fn needs_to_handle<'a>(_controller_opt: &'a ControllerOpt) -> Option<&'a Self::Resource> {
+        // TODO: fix compile error
+        // error[E0515]: cannot return value referencing temporary value
+        // if let Some(memory) = &controller_opt.resources.memory() {
+        //     return Some(memory);
+        // }
         None
```

Comment:
Is there a better way to fix this?

Reply:
I think you can remove this function.

Reply:
To answer your question, the reason compiler tells you is that you are checking a Option with reference inside, but you also return a new option with the same reference. That's not memory safe because now you have two Option containing the same reference. You can `clone` the value of `memory` and returns an entirely new option. However, I do find if we have to use clone in this case, we should question a bit what do we really want the code to do. In this case, it is just checking if the option is some and use the value. As you already did, we can safely to just not use this function and delete it.

Reply:
I understand now. Thank you.
Since this function is defined in the Controller trait, it needs to be removed from the trait definition.
I know it will be another amount of modification, but should I remove it within this PR?

Reply:
Wouldn't it be easier to do if this PR is merged?
https://github.com/containers/oci-spec-rs/pull/73

Reply:
Then don't remove it. Maybe put a TODO to explain.

Reply:
@guni1192 The above PR has been merged.

Reply:
@guni1192 cc: @yihuaf The timing was bad and this PR has been merged. Could you use another PR to handle it?